nova_vm/src/ecmascript/syntax_directed_operations/class_definitions.rs

```diff
@@ -1,40 +1,73 @@
-use crate::ecmascript::{abstract_operations::{operations_on_objects::construct, testing_and_comparison::is_constructor}, builtins::{ordinary::ordinary_create_from_constructor, ArgumentsList}, execution::{agent::ExceptionType, Agent, JsResult, ProtoIntrinsics}, types::{Function, InternalMethods, Object, Value}};
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use crate::{
+    ecmascript::{
+        abstract_operations::{
+            operations_on_objects::construct, testing_and_comparison::is_constructor,
+        },
+        builtins::{ordinary::ordinary_create_from_constructor, ArgumentsList},
+        execution::{agent::ExceptionType, Agent, JsResult, ProtoIntrinsics},
+        types::{Function, InternalMethods, Object, String, Value, BUILTIN_STRING_MEMORY},
+    },
+    engine::{
+        is_reference, CompileContext, CompileEvaluation, FunctionExpression, Instruction,
+        NamedEvaluationParameter, SendableRef,
+    },
+};
+use oxc_ast::{
+    ast::{self, MethodDefinitionKind},
+    syntax_directed_operations::{PrivateBoundIdentifiers, PropName},
+};
 
 pub(crate) fn base_class_default_constructor(
-        agent: &mut Agent,
-        _this_value: Value,
-        _: ArgumentsList,
-        new_target: Option<Object>,
-    ) -> JsResult<Value> {
+    agent: &mut Agent,
+    _this_value: Value,
+    _: ArgumentsList,
+    new_target: Option<Object>,
+) -> JsResult<Value> {
     // ii. If NewTarget is undefined, throw a TypeError exception.
     let Some(new_target) = new_target else {
-        return Err(agent.throw_exception_with_static_message(ExceptionType::TypeError, "class constructors must be invoked with 'new'"));
+        return Err(agent.throw_exception_with_static_message(
+            ExceptionType::TypeError,
+            "class constructors must be invoked with 'new'",
+        ));
     };
+
     // iii. Let F be the active function object.
     let _ = agent.running_execution_context().function.unwrap();
 
     // TODO: Figure out if it's actually okay to split this function into two,
     // or is F's [[ConstructorKind]] only decidable at runtime?
```

Comment:
I'm not actually sure if this is resolved: I've written it in as if it's okay to split these two but it may end up that it's not correct.

Essentially I think the question is if the default constructor of `Default` will change which branch it takes in these two new call cases:
```ts
class Base {
  constructor() {
    console.log("Not default");
  }
}

class Default extends Base {
}

class Derived extends Default {
  constructor() {
    super();
  }
}

new Default();
new Derived();
```

Reply:
I'm not sure how non-default constructors call the superclass constructor in the spec text, but for default constructors, it's called with `Construct`. And I think abstract closures are always expected to be called (or constructed) using the regular function call mechanisms, without magic, so `F` would be the current constructor, not the subclass's.

Reply:
Alright, I removed the comment.